actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -79,6 +71,117 @@ public void onReceive(Request request) throws Exception {
     }
   }
 
+  public void getCourse(Request request) {
+    String userId = (String) request.getRequest().get(JsonKey.USER_ID);
+
+    Map<String, Object> filter = new HashMap<>();
+    filter.put(JsonKey.USER_ID, userId);
+    filter.put(JsonKey.ACTIVE, ProjectUtil.ActiveStatus.ACTIVE.getValue());
+    SearchDTO searchDto = new SearchDTO();
+    searchDto.getAdditionalProperties().put(JsonKey.FILTERS, filter);
+
+    Map<String, Object> result =
+        ElasticSearchUtil.complexSearch(
+            searchDto,
+            ProjectUtil.EsIndex.sunbird.getIndexName(),
+            ProjectUtil.EsType.usercourses.getTypeName());
+    if (MapUtils.isNotEmpty(result)) {
+      addCourseDetails(request, result);
+    }
+    Response response = new Response();
+    response.put(JsonKey.COURSES, result.get(JsonKey.CONTENT));
+    sender().tell(response, self());
+  }
+
+  private void addCourseDetails(Request request, Map<String, Object> userCoursesResult) {
+    List<Map<String, Object>> batches =
+        (List<Map<String, Object>>) userCoursesResult.get(JsonKey.CONTENT);
+
+    if (CollectionUtils.isNotEmpty(batches)) {
+      return;
+    }
+    String requestBody = prepareCourseSearchRequest(batches);
+    if (StringUtils.isBlank(requestBody)) {
+      return;
+    }
+    ProjectLogger.log(
```

Comment:
Our logging follows a pttern ->

ProjectLogger.log(
          "UserManagementActor:getUserProfileV2:

Following logging pattern - makes easy to find sequence of calls - easier to debug